sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -295,6 +296,22 @@ object functions {
   def min(columnName: String): Column = min(Column(columnName))
 
   /**
+   * Aggregate function: returns the sample standard deviation of the values in a group.
+   *
+   * @group agg_funcs
+   * @since 1.5.0
+   */
+  def std(e: Column): Column = aggregate.Utils.standardDeviation(e.expr)
+
+  /**
+   * Aggregate function: returns the sample standard deviation of the values in a group.
+   *
+   * @group agg_funcs
+   * @since 1.5.0
+   */
+  def std(columnName: String): Column = std(Column(columnName))
```

Comment:
Call it `stddev`? So the name of the function matches the string in the describe result (btw, seems other databases also use `stddev`, which is an alias of `stddev_samp`).